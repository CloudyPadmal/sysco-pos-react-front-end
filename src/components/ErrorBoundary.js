import React from 'react';
import { Link } from 'react-router-dom'

class ErrorBoundary extends React.Component {

    constructor(props) {
        super(props);
        this.state = { hasError: false };
    }

    static getDerivedStateFromError(error) {
        // Update state so the next render will show the fallback UI.
        return { hasError: true };
    }

    render() {
        if (this.state.hasError) {
            return <h1 className="d-flex justify-content-center"
                style={{ height: '100%', margin: '30% 0 0 0' }}>
                <span className="badge badge-dark" style={{ padding: '50px' }}>
                    Cannot load the website ... <br /><br />Want to go back to
                    <br /><br /><Link to="/home">Home Page</Link>?</span>
            </h1>;
        }
        return this.props.children;
    }
}

export default ErrorBoundary;