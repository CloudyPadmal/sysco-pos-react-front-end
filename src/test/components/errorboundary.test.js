import React from 'react';
import { mountWrap } from '../enzyme';

import ErrorBoundary from '../../components/ErrorBoundary';

describe('<ErrorBoundary /> component', () => {

    it('renders with an error prompt', () => {
        const wrapper = mountWrap(
            <ErrorBoundary>
                <p>Hello</p>
            </ErrorBoundary>
        );
        wrapper.setState({
            hasError: true
        });
        expect(wrapper).toMatchSnapshot();
    });
    it('renders children when there is no error', () => {
        const wrapper = mountWrap(
            <ErrorBoundary>
                <p>Hello</p>
            </ErrorBoundary>
        );
        wrapper.setState({
            hasError: false
        });
        expect(wrapper).toMatchSnapshot();
    });
});