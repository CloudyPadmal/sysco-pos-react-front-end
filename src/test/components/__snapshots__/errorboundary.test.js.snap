// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorBoundary /> component renders children when there is no error 1`] = `
<ErrorBoundary>
  <p>
    Hello
  </p>
</ErrorBoundary>
`;

exports[`<ErrorBoundary /> component renders with an error prompt 1`] = `
<ErrorBoundary>
  <h1
    className="d-flex justify-content-center"
    style={
      Object {
        "height": "100%",
        "margin": "10% 0 0 0",
      }
    }
  >
    <span
      className="badge badge-dark"
      style={
        Object {
          "padding": "50px",
        }
      }
    >
      Cannot load the website ... 
      <br />
      <br />
      Want to go back to
      <br />
      <br />
      <Link
        replace={false}
        to="/home"
      >
        <a
          href="/home"
          onClick={[Function]}
        >
          Home Page
        </a>
      </Link>
      ?
    </span>
  </h1>
</ErrorBoundary>
`;
